tests/general/views/test_consolidated_profile_view.py

```diff
@@ -0,0 +1,566 @@
+        assert metrics == [
+            {'label': 'metric', 'value': 0.25, 'method': 'sibling'}
+        ]
+
+
+class TestParentLayersData(ConsolidatedProfileViewBase):
+
+    def test_version_parent_and_siblings(self):
+        version1, geography, hierarchy, profile = self.create_base_data(
+            "v2", None, self.hierarchy, self.profile
+        )
+
+        data = [
+            {"gender": "male", "age": "15", "count": 1},
+            {"gender": "female", "age": "16", "count": 2},
+        ]
+        indicator = self.create_versioned_data(
+            version1, geography, profile, data=data
+        )
+
+        response = self.get(
+            'all-details',
+            profile_id=profile.pk,
+            geography_code=geography.code,
+            data={
+                'version': 'v2',
+                'format': 'json'
+            },
+        )
+        self.assert_http_200_ok()
+
+        # assert parents
+        geo_data = response.data["profile"]["geography"]
+        assert len(geo_data["parents"]) == 2
+        assert geo_data["parents"][0]["code"] == self.root.code
+        assert geo_data["parents"][1]["code"] == self.geo1.code
+
+        # Assert parent & siblings and parent siblings
+        parent_layers = response.data["parent_layers"]
+        assert len(parent_layers) == 2
+        assert len(parent_layers[0]["features"]) == 1
+        assert parent_layers[0]["features"][0]["properties"]["code"] == self.geo1.code
+        assert len(parent_layers[1]["features"]) == 1
+        assert parent_layers[1]["features"][0]["properties"]["code"] == geography.code
+
+        # Add boundary for version & parent sibling
+        self.create_boundary(self.geo2, version1)
+        # Add sibling but with same version
+        self.create_boundary(self.dc1, version1)
+
+        self.tearDown()
+        response = self.get(
+            'all-details',
+            profile_id=profile.pk,
+            geography_code=geography.code,
+            data={
+                'version': 'v2',
+                'format': 'json'
+            },
+        )
+        self.assert_http_200_ok()
+
+        parent_layers = response.data["parent_layers"]
+        assert len(parent_layers) == 2
+        assert len(parent_layers[0]["features"]) == 2
+        assert parent_layers[0]["features"][0]["properties"]["code"] == self.geo1.code
+        assert parent_layers[0]["features"][1]["properties"]["code"] == self.geo2.code
+        assert len(parent_layers[1]["features"]) == 2
+        assert parent_layers[1]["features"][0]["properties"]["code"] == self.dc1.code
+        assert parent_layers[1]["features"][1]["properties"]["code"] == geography.code
+
+class TestChildrenData(ConsolidatedProfileViewBase):
+
+    def test_version_parent_and_siblings(self):
+        version1, geography, hierarchy, profile = self.create_base_data(
+            "v2", None, self.hierarchy, self.profile
+        )
```

Comment:
please document the geo hierarchy for each version in a comment for this test too.

Reply:
an alertnative to documenting the geo levels might be to name them in a way that conveys this, e.g. geo1a for root, geo1a2a and geo1a2b, and geo1a2b3a and so on.... or not :P